src/formatDate.js

```diff
@@ -8,7 +8,49 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const separator = toFormat[toFormat.length - 1];
```

Comment:
The assumption that the last element of `fromFormat` and `toFormat` is the separator may not always hold true. Consider explicitly specifying the separator or using a different method to determine it.